src/3rd_party/pathie-cpp/src/pathie.cpp

```diff
@@ -181,11 +181,10 @@ std::string Pathie::convert_encodings(const char* from_encoding, const char* to_
 std::string Pathie::utf8_to_filename(const std::string& utf8)
 {
   bool fs_encoding_is_utf8 = false;
-
+  char* fsencoding = NULL;
```

Comment:
Why was this moved up?

Reply:
`fsencoding` is defined only in the `else` part of the `ifdef`, resulting in a compilation error when this preprocessor path is taken. It's an oversight by the author, I had the same discussion with him when I proposed this change upstream.

Reply:
Thanks